python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -1764,15 +1747,20 @@ def test_dataframe_transpose(nulls, num_cols, num_rows, dtype):
     pdf = pd.DataFrame()
 
     null_rep = np.nan if dtype in ["float32", "float64"] else None
-
```

Comment:
Not really understanding the changes in this test -- can we add a comment to explain what's going on?

Reply:
Sure, so this was a change in behavior incase of `bool` column:

Until pandas `1.2.5`, setting a `nan` to bool column would result in conversion to `float`:
```python
s = pd.Series([True, False, True])
s
0     True
1    False
2     True
dtype: bool
s[[2]] = np.nan
s
0    1.0
1    0.0
2    NaN
dtype: float64
```

Now that's fixed to conversion to an `object` dtype:

```python
>>> s = pd.Series([True, False, True])
>>> s
0     True
1    False
2     True
dtype: bool
>>> s[[2]] = np.nan
>>> s
0     True
1    False
2      NaN
dtype: object
```

So in `cudf` we change `object` dtype to `str` type - for which there is no `transpose` implemented yet.

Hence the fix I have made in this pytest is to correctly cast the data to pandas nullable types and perform a transpose

Reply:
Ahh, thanks! Let's add a comment in the code here for our future selves?

Reply:
Done 👍